src/Button/Button.tsx

```diff
@@ -276,293 +279,296 @@ export const Button = React.forwardRef<HTMLElement, Props>(
-                    boxShadow: `0 1px 4px 0 rgba(18, 21, 26, 0.08), 0 0 0 2px ${
-                      theme === "light" ||
-                      color === defaultColor ||
-                      color === colors.white
-                        ? "#bbdbff"
-                        : getOffsetInPalette(Infinity, "lighter", color)
-                    }, inset 0 0 0 1px ${
-                      color === defaultColor || color === colors.white
-                        ? "#2075d6"
-                        : getOffsetInPalette(1, "darker", color)
-                    }, inset 0 -1px 0 0 rgba(18, 21, 26, 0.05)`,
-                  },
-                  "&:active, &[data-force-active-state], &[aria-expanded=true]": {
-                    ...(getTextColor({
-                      color,
-                      feel,
-                      theme,
-                      mode: ":hover",
-                    }) && {
-                      color: getTextColor({
-                        color,
-                        feel,
-                        theme,
-                        mode: ":active",
-                      }),
-                    }),
-
-                    backgroundColor:
-                      color === colors.white
-                        ? colors.white
-                        : feel === "raised"
-                        ? color
-                        : color === defaultColor
-                        ? theme === "dark"
-                          ? colors.grey.darker
-                          : colors.silver.base
-                        : getOffsetInPalette(2, "lighter", color),
-
-                    // The `box-shadow` properties are copied directly from Zeplin
-                    boxShadow:
-                      feel !== "flat"
-                        ? theme === "light"
-                          ? "inset 0 0 0 1px rgba(18, 21, 26, 0.2), inset 0 -1px 0 0 rgba(18, 21, 26, 0.05), inset 0 2px 2px 0 rgba(18, 21, 26, 0.12)"
-                          : "0 0 0 1px rgba(18, 21, 26, 0.2), 0 1px 4px 0 rgba(18, 21, 26, 0.08), 0 -1px 0 0 rgba(18, 21, 26, 0.16), inset 0 1px 2px 0 rgba(18, 21, 26, 0.42)"
-                        : "none",
-                    outline: "0",
-                  },
-                },
-              ])
-            )
-          ),
-          disabled,
-          onClick: (event: React.MouseEvent<HTMLElement, MouseEvent>) => {
-            if (disabled) return event.preventDefault();
 
-            if (otherProps.onClick) {
-              otherProps.onClick(event);
-            }
+          const propsToPass = mergeProps(
+            // Omit `onClick` from `otherProps` beacuse we'll be conditionally
+            // calling it in the `onClick` handler depending on the `disabled`
+            // prop.
+            omit(otherProps, "onClick"),
+            mergeProps(focusProps, {
+              ref,
+              className: classnames(
+                isFocusVisible && cx(css(focusedStyles)),
```

Comment:
this `cx` seems redundant?

Reply:
I had no idea what `cx` was doing; this is totally using it wrong everywhere! I have another commit I'm going to push that shrinks down the `<ClassNames>` insides to only relate to stuff that actually needs `ClassNames` and I'll use `cx` correctly.